src/config/grpc_fetch.rs

```diff
@@ -0,0 +1,176 @@
+use anyhow::{Context, Result};
+use base64::prelude::BASE64_STANDARD;
+use base64::Engine;
+use hyper::Method;
+use nom::AsBytes;
+use prost::Message;
+use prost_reflect::prost_types::{FileDescriptorProto, FileDescriptorSet};
+use serde_json::Value;
+
+use crate::grpc::protobuf::ProtobufSet;
+use crate::runtime::TargetRuntime;
+
+const REFLECTION_PROTO: &str = include_str!(concat!(
+    env!("CARGO_MANIFEST_DIR"),
+    "/proto/reflection.proto"
+));
+
+/// This function is just used for better exception handling
+fn get_protobuf_set() -> Result<ProtobufSet> {
+    let descriptor = protox_parse::parse("reflection", REFLECTION_PROTO)?;
+    let mut descriptor_set = FileDescriptorSet::default();
+    descriptor_set.file.push(descriptor);
+    ProtobufSet::from_proto_file(&descriptor_set)
+}
+
+/// Makes `ListService` request to the grpc reflection server
+pub async fn list_all_files(url: &str, target_runtime: &TargetRuntime) -> Result<Vec<String>> {
+    let protobuf_set = get_protobuf_set()?;
+
+    let mut methods = vec![];
+    let mut url: url::Url = url.parse()?;
+    url.set_path("grpc.reflection.v1alpha.ServerReflection/ServerReflectionInfo");
+
+    let mut req = reqwest::Request::new(Method::POST, url);
+    *req.body_mut() = Some(reqwest::Body::from(b"\0\0\0\0\x02:\0".to_vec())); // magic :)
+
+    let resp = target_runtime.http.execute(req).await?;
+    let body = resp.body.as_bytes();
```

Comment:
instead of using raw reqwest and create body manually consider using `TargetRuntime` and `grpc::RequestTemplate`.

For RequestTemplate use the `protobuf_set` you've already created and its `convert_input` method to construct bytes for body from the json string. 
How it works you can find in file `src/grpc/protobuf.rs` and its tests.
e.g. your `b"\0\0\0\0\x02:\0"` is the same as `protobuf_set.convert_input("{listServices: ""}")`